### ISSUE TITLE
`mode: 'jit'` with `2.1.0` causing issues.

### ISSUE THREAD

**Issue by crswll** [tailwindlabs/tailwindcss-intellisense#293]
Hi!

When I use `mode: 'jit'` I get the following error in `tailwindcss-intellisense` output.

```bash
Found Tailwind config file: /Users/bcrisw467/www/project-name/tailwind.config.js
Found tailwindcss v2.1.0: /Users/bcrisw467/www/project-name/node_modules/tailwindcss

warn - You have enabled the JIT engine which is currently in preview.
warn - Preview features are not covered by semver, may introduce breaking changes, and can change at any time.
Failed to initialise: TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received undefined
    at validateString (internal/validators.js:120:11)
    at Url.parse (url.js:159:3)
    at Object.urlParse [as parse] (url.js:154:13)
    at trackModified (/Users/bill/www/project-name/node_modules/tailwindcss/jit/lib/setupContext.js:230:24)
    at /Users/bill/www/project-name/node_modules/tailwindcss/jit/lib/setupContext.js:726:38
    at /Users/bill/www/project-name/node_modules/tailwindcss/jit/index.js:35:47
    at LazyResult.runOnRoot (/Users/bill/www/project-name/node_modules/postcss/lib/lazy-result.js:303:16)
    at LazyResult.runAsync (/Users/bill/www/project-name/node_modules/postcss/lib/lazy-result.js:355:26)
    at LazyResult.async (/Users/bill/www/project-name/node_modules/postcss/lib/lazy-result.js:205:30)
    at LazyResult.then (/Users/bill/www/project-name/node_modules/postcss/lib/lazy-result.js:190:17) {
  code: 'ERR_INVALID_ARG_TYPE'
```

**Comment by timothyjoelwright** [comment]
I'm also having this issue, although my output suggests a TypeError:

```
warn - You have enabled the JIT engine which is currently in preview.
warn - Preview features are not covered by semver, may introduce breaking changes, and can change at any time.
Failed to initialise: TypeError: Cannot read property 'map' of undefined
    at /Users/timothyjoelwright/Documents/Method.Foundation/web/node_modules/tailwindcss/jit/lib/setupContext.js:783:9
    at /Users/timothyjoelwright/Documents/Method.Foundation/web/node_modules/tailwindcss/jit/index.js:35:47
    at LazyResult.runOnRoot (/Users/timothyjoelwright/Documents/Method.Foundation/web/node_modules/postcss/lib/lazy-result.js:303:16)
    at LazyResult.runAsync (/Users/timothyjoelwright/Documents/Method.Foundation/web/node_modules/postcss/lib/lazy-result.js:355:26)
    at LazyResult.async (/Users/timothyjoelwright/Documents/Method.Foundation/web/node_modules/postcss/lib/lazy-result.js:205:30)
    at LazyResult.then (/Users/timothyjoelwright/Documents/Method.Foundation/web/node_modules/postcss/lib/lazy-result.js:190:17)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

It initializes successfully when `mode: jit` is removed from `tailwind.config.js`

**Comment by crswll** [comment]
The error I mentioned was fixed. The one @timothyjoelwright mentioned is because `mode: 'jit'` isn't compatible yet but is being worked on from what I hear.

**Comment by gianmd** [comment]
do we have an ETA to make it work with the latest version and jit mode enabled?

**Comment by codemonkey800** [comment]
I got a workaround working by using an environment variable to determine the mode for the Tailwind config:

```js
let mode;
if (process.env.TAILWIND_JIT) {
  mode = 'jit';
}

module.exports = {
  mode,
  // ...,
};
```

in your `package.json`, you can do something like:

```json
{
  "scripts": {
    "dev": "TAILWIND_JIT=true next dev"
  }
}
```

Although, a proper solution would be great 😄

**Comment by crswll** [comment]
Brad mentioned a quick fix that basically makes it work like you're not using `jit` before it's done properly. I don't know when to expect it but sure it's on the radar.

**Comment by bradlc** [comment]
This is fixed in `v0.5.10` – note that this release does not include full support for the JIT engine. That is coming very soon though! 👀 For now the extension behaves as if you don't have JIT enabled.
